Release 5.1.0

ParaskP7 opened the pull request:
Release process PR following [releasing instructions](https://github.com/Automattic/a8c-ci-toolkit-buildkite-plugin?tab=readme-ov-file#releasing).

Related PRs:

- #164 

FYI: Will admin-merge once CI allows (unless already approved) and proceed with creating a GitHub release from it.

ParaskP7 committed:
Release: Update changelog and readme for version 5.1.0

AliSoftware commented on CHANGELOG.md:
```diff
@@ -48,6 +48,12 @@ _None._
+
+### New Features
+
+- `AndroidManifest`: Merged Manifest Diff [#164]
```
Can we maybe provide a more explicit entry for this CHANGELOG after all? Doesn't feel super descriptive and kinda abstract to me

AliSoftware replied:
Like maybe at least mention the _name_ of the new command this it introduced (namely `comment_with_manifest_diff`) as part of this entry

ParaskP7 replied:
👍 and how about this (9646fa989b0e59320f39bdd860ab3e996e0b93bc) @AliSoftware ? 🙏

ParaskP7 committed:
Release: Add a more explicit/descriptive entry for merged manifest diff

PR Comment: https://github.com/Automattic/
a8c-ci-toolkit-buildkite-plugin/pull/169#discussion_r1981484130

AliSoftware commented on CHANGELOG.md:
```diff
@@ -48,6 +48,12 @@ _None._
+
+### New Features
+
+- Add a script for commenting manifest diff on PRs in Android projects [#164]
```
```suggestion
- Add new `comment_with_manifest_diff` command for commenting manifest diff on PRs in Android projects [#164]
```

ParaskP7 committed:
Release: Add a more explicit/descriptive entry for merged manifest diff

Co-authored-by: Olivier Halligon <olivier.halligon@automattic.com>

mokagio commented:
Took the liberty of restarting the failed Lint and Test CI steps, which failed due to the out of space issue that @iangmaia and @AliSoftware addressed via cleaning up the agents and shipping the `nvm` plugin v0.5.0.

Now CI is all green and I might as well merge and tag while I'm here...

ParaskP7 commented:
Thanks for taking care of that @mokagio ! 🚀 

> ...which failed due to the out of space issue that @iangmaia and @AliSoftware addressed via cleaning up the agents and shipping the nvm plugin v0.5.0.

I am so glad this is now resolved! 💪
